src/mlpack/methods/ann/layer/embedding.hpp

```diff
@@ -0,0 +1,157 @@
+/**
+ * @file methods/ann/layer/embedding.hpp
+ * @author Marcus Edel
+ * @author Kumar Utkarsh
+ *
+ * Definition of the Embedding (embedding) layer class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_EMBEDDING_HPP
+#define MLPACK_METHODS_ANN_LAYER_EMBEDDING_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/methods/ann/regularizer/no_regularizer.hpp>
+
+#include "layer.hpp"
+
+namespace mlpack {
+
+/**
+ * The Embedding class stores word embeddings and retrieves them using tokens.
+ * It must always be the first layer of any network it is used in.
+ *
+ * The input to the Embedding class is a matrix of shape (sequenceLength,
+ * batchSize). The matrix consists of tokens which are used to lookup the table
+ * (i.e. weights) to find the embeddings of those tokens.
+ *
+ * The input shape : (sequenceLength, batchSize).
+ * The output shape : (sequenceLength * embeddingSize, batchSize).
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template<
+    typename MatType = arma::mat,
+    typename RegularizerType = NoRegularizer
+>
+class Embedding : public Layer<MatType>
+{
+ public:
+  using CubeType = typename GetCubeType<MatType>::type;
+
+  // Create an Embedding layer.
+  Embedding();
+
+  /**
+   * Create the Embedding layer object with the specified number of output
+   * dimensions.  The vocabulary size (number of possible inputs) and the
+   * embedding size (dimensionality of the output) must be given.
+   *
+   * @param outSize The output dimension.
+   * @param regularizer The regularizer to use, optional (default: no
+   *     regularizer).
+   */
+  Embedding(const size_t vocabSize,
```

Comment:
The documentation does not match the parameters.

Reply:
Fixed in ab1d2460f6.